core/unix/os.c

```diff
@@ -8651,16 +8663,14 @@ post_system_call(dcontext_t *dcontext)
             size_t sizemask;
         } data_t;
         data_t *data = (data_t *)sys_param(dcontext, 5);
-        kernel_sigset_t app_sigblocked =
-            handle_post_extended_syscall_sigmasks(dcontext, success);
+        handle_post_extended_syscall_sigmasks(dcontext, success);
         safe_write_ex((void *)&data->sigmask, sizeof(kernel_sigset_t),
-                      (const void *)&app_sigblocked, NULL);
+                      (const void *)dcontext->sys_param4, NULL);
```

Comment:
bug: above you set sys_param4 to sizemask: looks like you want the mask itself, not the size.

How does this pass the test you added?  Am I missing something?

Reply:
Fixed.